Improve/fix registration status icon text

kr-matthews opened the pull request:
This competition is in just over a month. Previously it also said "in 1 month".

![image](https://github.com/user-attachments/assets/9e41da9b-f7f0-4c21-a411-c7cc8b708a8d)

This competition is tomorrow. Previously it would have said "in 2 hours".

![image](https://github.com/user-attachments/assets/6d30e1a9-d54d-4bef-b1d6-e84082c81fd3)

Unfortunately Luxon doesn't seem to be using the locale, so that will need to be fixed if possible (in another PR). Also, other languages have the extra "in" too, it seems (and missing periods).

![image](https://github.com/user-attachments/assets/621c6102-9250-40bd-aa9e-418da9fb3020)

kr-matthews commented:
Actually these strings are also used in the 'all competitions' page (both the old and new UI), so this PR just shifts the extra 'in' problem. However, that page is correctly translating the 'days' message. So we probably want a consistent approach between the two - using luxon in both (with correct locale), or explicitly calculating the number of days and directly translating that (note this never would show "1 month", it's always days). For the former approach, I'm not sure how to make those changes in ruby, so someone else would have to pick this up.

gregorbg commented on app/webpacker/components/MyCompetitions/UpcomingCompetitionTable.jsx:
```diff
@@ -23,11 +23,18 @@ const competingStatusIcon = (competingStatus) => {
+
   if (competition.registration_status === 'not_yet_opened') {
-    return I18n.t('competitions.index.tooltips.registration.opens_in', { duration: DateTime.fromISO(competition.registration_open).toRelative() });
+    return I18n.t('competitions.index.tooltips.registration.opens_in', { duration: DateTime.fromISO(competition.registration_open).toRelative(toRelativeOptions) });
```
I think in terms of Luxon respecting the locale, it's (supposed to be) as simple as passing the locale ISO code to `fromISO` like so: `DateTime.fromISO(dateIsoString, { locale: 'fr' })`. Cf. https://moment.github.io/luxon/#/intl?id=setting-the-locale

We expose the currently seleted locale as `I18n.locale` on the same I18n object that also does `t` and `tArray`.

gregorbg replied:
Interestingly, even the `toRelativeOptions` object supports a `locale` key.

kr-matthews replied:
Fixed by a separate PR.

gregorbg commented on app/webpacker/components/MyCompetitions/UpcomingCompetitionTable.jsx:
```diff
@@ -23,11 +23,18 @@ const competingStatusIcon = (competingStatus) => {
   }
   if (competition.registration_status === 'past') {
-    return I18n.t('competitions.index.tooltips.registration.closed', { days: DateTime.fromISO(competition.start_date).toRelative() });
+    return I18n.t('competitions.index.tooltips.registration.closed', { days: DateTime.fromISO(competition.start_date).toRelative(toRelativeOptions) });
```
See above for passing locale information to Luxon

gregorbg reviewed (commented):
Added some comments about locale handling

gregorbg commented:
About the handling of `in`: This is a "devil or the deep blue sea" scenario. Luxon does have `luxonDt.diffNow().toHuman()` method, which (with a little bit of fiddling in terms of passing in the correct `config` options) would boild down to `3 days, 8 hours` without the "in".

However, for most languages there's a notable difference because of grammar. In German for example, if you just enumerate the time units in a "raw" fashion, it's `6 Monate, 3 Tage` (six months and three days). In case of a relative time difference, as in "the event happens **in 6 months and 3 days**" you'd have to change the declension to `in 6 Monate**_n_** und 3 Tage**_n_**`.

So I personally feel like it's best to let the Luxon library handle the "in" and "ago" and nasty declensions, and remove them from our translatable I18n keys like you did here.

kr-matthews commented:
Another PR fixed the wrong locale issue.
I've updated this PR to take that into account. And instead of 'fixing' the existing strings, I've created new ones. Because the existing strings are also used elsewhere, and there the "in" is required. Alternatively, we could update those instances to handle "in" differently and not introduce any new strings.

gregorbg commented:
> Alternatively, we could update those instances to handle "in" differently and not introduce any new strings.

I would prefer that, if it's not too much to ask. When you get started and you realize you're wasting more than say ~45 minutes and you have to dig into the source code of some old. deprecated Ruby library, then stop what you're doing, post another comment here and I'll approve the additional transations :wink:

kr-matthews commented:
> > Alternatively, we could update those instances to handle "in" differently and not introduce any new strings.
> 
> I would prefer that, if it's not too much to ask. When you get started and you realize you're wasting more than say ~45 minutes and you have to dig into the source code of some old. deprecated Ruby library, then stop what you're doing, post another comment here and I'll approve the additional transations 😉

The only ruby use is in the old competitions overview code. We recently switched over to the new react version. Can we just remove that old code now? I'm under the impression we won't be switching back to it again.

kr-matthews commented:
I reverted to the existing strings and updated them. The old competitions overview ruby code will be wrong, but it doesn't matter since it's being removed and can't be seen in the meantime anyway.

I also adjusted it so that for registration open it is precise (since it's a date time), but for competition start the precision is at most days (since it's a date, not a date time).

![wst-c](https://github.com/user-attachments/assets/29124b62-0412-448c-88b6-5055c3d33263)
![wst-b](https://github.com/user-attachments/assets/a8f234fa-de0a-43e7-a335-19da365eac50)
![wst-1](https://github.com/user-attachments/assets/f2307fec-3f41-4d08-b00b-108aeacd15b5)
![wst-a](https://github.com/user-attachments/assets/a0c86200-adcc-4979-a7ea-350060173bbe)
![wst-A](https://github.com/user-attachments/assets/a3b8682f-6c11-423d-a025-60af4b607cc2)
![wst-B](https://github.com/user-attachments/assets/922b1cfa-bf97-44c5-9535-0afdb9a06e82)
![wst-C](https://github.com/user-attachments/assets/cd1efe66-9012-455a-a8a4-080def66f8e3)
![wst-D](https://github.com/user-attachments/assets/c72ee56e-03ff-4e31-86e9-e3999a1fa0aa)
![wst-3](https://github.com/user-attachments/assets/5cc396be-9903-42cf-9e2b-f007d39bf36d)
![wst-2](https://github.com/user-attachments/assets/4a707721-3946-421a-ac86-c13b773e6bca)

kr-matthews commented:
Can we merge this? Or are we waiting for the other PR which removes the old code, to be safe?

gregorbg commented:
I went ahead and merged the other PR, so this is now good to :ship:
